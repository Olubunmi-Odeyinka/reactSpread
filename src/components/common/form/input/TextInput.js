import React from 'react';
import PropTypes from 'prop-types';
import {getFieldNameFromName } from '../../../../utilities/fieldName';

const TextInput = ({name, type, rowValue, onChange, readOnly}) => {

  return (
    <div className='form-group'>
      <div className="field">
        <input
          type={type || 'text'}
          name={name}
          value={rowValue[getFieldNameFromName(name)]}
          onChange={onChange}
          readOnly={readOnly}
          className="form-control"/>
      </div>
    </div>
  );
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  touched: PropTypes.bool,
  error: PropTypes.string,
  focus: PropTypes.bool,
  readOnly: PropTypes.bool
};

export default TextInput;
