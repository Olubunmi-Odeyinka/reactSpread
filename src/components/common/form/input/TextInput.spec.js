import React from 'react';
import { shallow } from 'enzyme';
import TextInput from './TextInput';

let props, wrapper;
beforeEach(() => {
  props = {
    name: 'Description',
    label: 'Description',
    rowValue: '',
    readOnly: false,
    onChange: jest.fn()
  };
  wrapper = shallow(<TextInput {...props} />);
});

test('The text input component input component must render properly', ()=>{

  expect(wrapper).toMatchSnapshot();
});

// test("The text input component label should be 'Description' as set in props", ()=>{
//   expect(wrapper.find('label').text()).toEqual('Description');
// });
//
// test("The text input component name should be 'Description' as set in props", ()=>{
//   expect(wrapper.find('Field').props().name).toBe('Description');
// });
//
// test("The text input component readOnly should be 'false' as set in props", ()=>{
//   expect(wrapper.find('Field').props().readOnly).toBe(false);
// });
//
// test("The text input component autoFocus should be 'true' as set in props", ()=>{
//   expect(wrapper.find('Field').props().readOnly).toBe(false);
// });
