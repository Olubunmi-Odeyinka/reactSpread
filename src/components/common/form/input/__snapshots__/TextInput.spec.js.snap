// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The text input component input component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    label="Description"
    name="Description"
    onChange={[Function]}
    readOnly={false}
    rowValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="field"
>
        <input
                className="form-control"
                name="Description"
                onChange={[Function]}
                readOnly={false}
                type="text"
                value={undefined}
        />
</div>,
      "className": "form-group",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="form-control"
          name="Description"
          onChange={[Function]}
          readOnly={false}
          type="text"
          value={undefined}
/>,
        "className": "field",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "name": "Description",
          "onChange": [Function],
          "readOnly": false,
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="field"
>
          <input
                    className="form-control"
                    name="Description"
                    onChange={[Function]}
                    readOnly={false}
                    type="text"
                    value={undefined}
          />
</div>,
        "className": "form-group",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control"
            name="Description"
            onChange={[Function]}
            readOnly={false}
            type="text"
            value={undefined}
/>,
          "className": "field",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "Description",
            "onChange": [Function],
            "readOnly": false,
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
